core/src/main/java/org/jobrunr/jobs/details/JobDetailsBuilder.java

```diff
@@ -71,6 +71,13 @@ public final LinkedList<Object> getStack() {
     }
 
     public JobDetails getJobDetails() {
+        if (jobDetailsMethodName == null) {
+            throw new IllegalStateException("Could not determine method name from lambda bytecode. This should not happen. Please report this as a bug with as much details as you can.");
+        }
+        if (jobDetailsMethodName.endsWith("$default") && classExists("kotlin.KotlinVersion")) {
+            throw new IllegalArgumentException("Unsupported lambda", new UnsupportedOperationException("You are (probably) using Kotlin default parameter values which is not supported by JobRunr."));
+        }
```

Comment:
I thought these would be better here but for some reason `getJobDetails` is called then `JobDetailsInstruction` is called and sets the proper method.